tools/ClangTidyAll.sh

```diff
@@ -8,8 +8,53 @@ ALL_SOURCE_FILES=`find \
                   @CMAKE_SOURCE_DIR@/tests \
                   -name "*.cpp"`
 
-for file in $ALL_SOURCE_FILES
+if [ ${#ALL_SOURCE_FILES[@]} == 0 ]; then
+   echo "No C++ files found in repository"
+   exit 0
+fi
+
+BUILD_DIR=@CMAKE_BINARY_DIR@
+
+if [ $# != 0 ] && [ $# != 1 ]; then
+    echo "Wrong number of arguments passed to ClangTidyAll.sh, $#"
+    echo "Expecting none or just the number of threads to run on."
+    exit 1
+fi
+
+THREADING_FLAG="-j 1"
+if [ $# == 1 ]; then
+    THREADING_FLAG="-j $1"
+fi
+
+# Try to find run-clang-tidy-LLVM_VERSION in the path.
+RUN_CLANG_TIDY_BIN=
+for dir in $(echo ${PATH} | tr ':' '\n');
 do
-    printf "\n\nChecking file: $file\n"
-    make clang-tidy FILE=$file
+    if [ -d $dir ]; then
+        RUN_CLANG_TIDY_BIN=`find $dir -name run-clang-tidy* | head -n 1`
+        if [ ! -z "${RUN_CLANG_TIDY_BIN}" ]; then
+            break
+        fi
+    fi
 done
+
+EXIT_CODE=0
+
+# Check for run-clang-tidy, then run-clang-tidy-LLVM_VERSION, and
+# if we can't find those, then just loop over the files one at a time.
+if command -v run-clang-tidy > /dev/null 2>&1; then
```

Comment:
Might be good to check `run-clang-tidy.py` explicitly as well.  You can turn the repeated checks into a loop like
```
for run-clang-tidy in run-clang-tidy run-clang-tidy.py "${RUN_CLANG_TIDY_BIN}" ;
do
    if command -v "${run-clang-tidy}" >/dev/null 2>&1; then
etc.
```

I guess you'll need a flag to stop when you've found a working one.  Same for the other file.

Reply:
Agree on checking run-clang-tidy.py and putting into a loop with a `break` and a flag if you find one (the flag so you know whether to run 'make clang-tidy').